stdlib/public/Concurrency/Actor.cpp

```diff
@@ -1894,11 +1926,19 @@ static void swift_job_run_on_serial_and_task_executorImpl(Job *job,
   }
 }
 
+// FIXME(concurrency): Can we remove this method during Swift 6 before 6.0 is released? (or make it crash)
+// DEPRECATED. This runtime entry point is incorrect and WILL result in dropping isolation guarantees.
+// This entry point backs the deprecated/banned `runSynchronously(on executor: UnownedTaskExecutor)` API,
+// but instead Swift MUST call `job.runSynchronously(isolatedTo: UnownedSerialExecutor, taskExecutor:  UnownedTaskExecutor)
 SWIFT_CC(swift)
 static void swift_job_run_on_task_executorImpl(Job *job,
                                                TaskExecutorRef taskExecutor) {
```

Comment:
This is an entrypoint from Swift that is incorrect. I'm keeping it around but if we could remove it that would be best, along with killing the `runSynchronously(on executor: UnownedTaskExecutor)`